test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -179,8 +190,9 @@ class ClusterManagerImplTest : public testing::Test {
   Router::ContextImpl router_context_;
 };
 
-MATCHER_P(CustomDnsResolversSizeEquals, expectedResolvers, "") {
-  return expectedResolvers.size() == arg.size();
+// Compare the expected protobuf message matches with typed_dns_resolver_config parameter.
+MATCHER_P(CustomTypedDnsResolverConfigEquals, expectedTypedDnsResolverConfig, "") {
```

Comment:
Is this matcher really needed? You may be able to use the `ProtoEq` matcher from `test/test_common/utility.h`

Reply:
That's right. Let me use ProtoEq directly. Thanks!